src/functions/retryProcessing/application/RetryProcessor.ts

```diff
@@ -25,6 +26,7 @@ export class RetryProcessor implements IRetryProcessor {
 
   async processSuccessful(): Promise<number> {
     try {
+      await this.connection.promise().query(setIsolationLevelSerializable);
```

Comment:
Transaction isolation level is a Session variable - once set, it applies to all following transactions on the same connection - so you just need to call it once at the beginning of the processing (probably where the connection is being created in the first place)

Reply:
Have moved the query to where the connection is being established to remove sending duplicate queries.